src/Features/CSharp/Portable/CodeRefactorings/SyncNamespace/CSharpChangeNamespaceService.cs

```diff
@@ -135,89 +192,163 @@ protected override SyntaxList<MemberDeclarationSyntax> GetMemberDeclarationsInCo
+                        members = members.Replace(last, lastWithTrivia);
+                    }
+                    else
+                    {
+                        eofToken = eofToken.WithPrependedLeadingTrivia(
+                            namespaceOpeningTrivia.Concat(namespaceClosingTrivia));
+                    }
+
+                    // Moving inner imports out of the namespace declaration can lead to a break in semantics.
+                    // For example:
+                    //
+                    //  namespace A.B.C
+                    //  {
+                    //    using D.E.F;
+                    //  }
+                    //
+                    //  The using of D.E.F is looked up with in the context of A.B.C first. If it's moved outside,
+                    //  it may fail to resolve.
+
+                    return root.Update(
+                        root.Externs.AddRange(namespaceDecl.Externs),
+                        root.Usings.AddRange(namespaceDecl.Usings),
+                        root.AttributeLists,
+                        root.Members.ReplaceRange(namespaceDecl, members),
+                        eofToken);
+                }
+
+                // Change namespace name
+                return root.ReplaceNode(
+                    namespaceDecl, 
+                    namespaceDecl.WithName(
+                        CreateNameSyntax(targetNamespaceParts, aliasQualifier: null, targetNamespaceParts.Length - 1)
+                        .WithTriviaFrom(namespaceDecl.Name).WithAdditionalAnnotations(WarningAnnotation))
+                        .WithoutAnnotations(ContainerAnnotation));
             }
 
-            // We should have a single member which is a namespace declaration in this compilation unit.
-            var namespaceDeclaration = compilationUnit.DescendantNodes().OfType<NamespaceDeclarationSyntax>().Single();
+            throw ExceptionUtilities.Unreachable;
+        }
+
+        /// <summary>
+        /// For the node specified by <paramref name="span"/> to be applicable container, it must be a namespace 
+        /// declaration or a compilation unit, contain no partial declarations and meet the following additional
+        /// requirements:
+        /// 
+        /// - If a namespace declaration:
+        ///    1. It doesn't contain or is nested in other namespace declarations
+        ///    2. The name of the namespace is valid (i.e. no errors)
+        ///
+        /// - If a compilation unit (i.e. <paramref name="span"/> is empty), there must be no namespace declaration
+        ///   inside (i.e. all members are declared in global namespace)
+        /// </summary>
+        protected override async Task<SyntaxNode> TryGetApplicableContainerFromSpanAsync(Document document, TextSpan span, CancellationToken cancellationToken)
+        {
+            var compilationUnit = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false) as CompilationUnitSyntax;
```

Comment:
just cast.  'as' indicates you think it might not be this type and want to check for null.  so an 'as' should always have a null-check, otherwise, you shoudl be casting.

Reply:
Fixed